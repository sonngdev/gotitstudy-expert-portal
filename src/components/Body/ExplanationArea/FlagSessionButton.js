import React from 'react';
import { ButtonWithModal } from '../../shared';

export function FlagSessionButton() {
  return (
    <ButtonWithModal
      buttonProps={{
        className: 'flag-session',
        variant: 'negative',
      }}
      buttonAction="Flag session"
      modalBody="You are flagging this chat session as inappropriate. Your student will be notified and this session will be terminated! Do you want to continue?"
      onConfirm={() => window.location.reload()}
    />
  );
}
