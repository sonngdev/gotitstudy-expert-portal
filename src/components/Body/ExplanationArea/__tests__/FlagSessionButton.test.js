import React from 'react';
import { shallow } from 'enzyme';
import { FlagSessionButton } from '../FlagSessionButton';

describe('<FlagSessionButton />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FlagSessionButton />);
  });

  it('renders', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('reloads window when modal is confirmed', () => {
    const { location } = window;
    delete window.location;
    window.location = { reload: jest.fn() };

    const confirm = wrapper.dive().find('Modal').find('.confirm');
    confirm.simulate('click');
    expect(window.location.reload).toHaveBeenCalled();

    window.location = location;
  });
});
