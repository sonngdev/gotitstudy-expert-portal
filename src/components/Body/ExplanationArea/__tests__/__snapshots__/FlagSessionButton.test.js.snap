// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlagSessionButton /> renders 1`] = `"<ButtonWithModal buttonProps={{...}} buttonAction=\\"Flag session\\" modalBody=\\"You are flagging this chat session as inappropriate. Your student will be notified and this session will be terminated! Do you want to continue?\\" onConfirm={[Function: onConfirm]} modalSize=\\"small\\" modalTitle=\\"\\" confirmButtonText=\\"\\" />"`;
